### PR TITLE
chore(edit-content): Merge edit content and fields libraries

### LINKED ISSUE #26634: Edit Content: Move Binary Field to Edit Content Lib
### Parent Issue

[#26628](https://github.com/dotCMS/core/issues/25445)

### Task

Move the Binary Field to `edit-content` lib and delete the `contenttype-fields` lib.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

The Binary Field component should be in `edit-content` lib.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by rjvelazco**
### <samp>🤖 Generated by Copilot at 01e0f9e</samp>

Refactored the binary field component to have a separate implementation for the edit content mode. Moved the component files from the `@dotcms/contenttype-fields` library to the `@dotcms/edit-content` library and renamed them accordingly. Updated the imports and the templates of the components that use the binary field component. Deleted the unused `@dotcms/contenttype-fields` library.

## Related Issue
Fixes #26634

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 01e0f9e</samp>

*  Replace `DotBinaryFieldComponent` with `DotEditBinaryFieldComponent` in `AppModule` and `DotEditContentFieldComponent` to use the wrapper component for the binary field in the edit content mode ([link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769L9-R10), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769L22-R22), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769L32-R32), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-685da2f87189fff825b7aec2666cfbabdb33c40f8f9d8c82df77ebb8652ccd4aL50-R50), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-23d2e2fd9b71e1819e10ae720cb79ecf5e4d75e51fee641b020017fe8186640fL37-R36), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-f03434ad88316542a23c3fb0bf50b6650130632c3e7953d7ee3c882cb5950d06L32-R32))
* Export `DotEditBinaryFieldComponent` from `@dotcms/edit-content` library to make it accessible for other modules ([link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-8e4ebf6e3bd0e536ddd7567da45f2092c49a46801b2e8cfa22531cfd312d137fR2))
* Update imports of `DotBinaryFieldComponent` and `DotEditBinaryFieldComponent` in `DotEditContentFieldComponent` and its spec file to use the relative path for the latter ([link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-23d2e2fd9b71e1819e10ae720cb79ecf5e4d75e51fee641b020017fe8186640fL8-R10), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-f03434ad88316542a23c3fb0bf50b6650130632c3e7953d7ee3c882cb5950d06L5-R8))
* Change type of `FIELD_TYPES_COMPONENTS` constant in `DotEditContentFieldComponent` spec file to `Partial<Record<FIELD_TYPES, Type<unknown>>>` to make it more accurate and flexible ([link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-23d2e2fd9b71e1819e10ae720cb79ecf5e4d75e51fee641b020017fe8186640fL25-R24))
* Rename and move `DotBinaryFieldEditorComponent` and `DotBinaryFieldPreviewComponent` to `DotEditBinaryFieldEditorComponent` and `DotEditBinaryFieldPreviewComponent` respectively and update their imports and spec files accordingly ([link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-f272a7bf8dc37684d08819c22613fe15e02bc532a616b0d34050321c785b0fba), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-78ab8437e4ead907e6a9de800aec3b83741b0ed82be52f301274fdc19c0e61a8), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-7fe2497c0f836b6e53e7e9296d3ea59f6b6d1c5733cfaa25fe59705a32a6e9deL15-R16), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-dff7597c4a9474a0aecf888359205481db79b62fdc6c2422335f333fc1d78b11), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-da9647f7218b4bcceb4867a577efa8d3f93dd749eb59dc5edcf79470f996c842))
* Delete `contenttype-fields` library and its files as it is no longer needed ([link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-cacac328a342cb16fb2af9bbf6cb4fa6197c1c8904e92d59a6570320e64632c1), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-f6c5d698cb33052a85cc6e55a6cc19fc6b9946b8d7d050448a185235f459f3d6), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-d2ac97231dfb24ee69701ed1eb85d4ee075e1d51eeeca8d7b4d7458827b61b9c), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-fa7cb5c3537ef1e3ba56436c1dca626f34c7eb4e2f9f6bf782df266c29279598), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-810dde5b6a6075391e20731b0055efcaefa32e6a8f0d5a26f45973084ca94b3f), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-6e9f82e7f0d7e1ef575bd32528529f1e4a734323df2657d9929136294c398bfc), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-6acb7473004e8fdd0dcfdc0c3e9e7782bc71c54229f93d8acd2062d4f0f8b32c), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-e8d41a957b9171e019cfd0fa1f3e8ec76d5abea3239b41b6fba5a2ee1f581c41), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-e12f6b5862e180d0f5e98874d1817ad7cf7c0b9998a1330aac77f2748f2930b7), [link](https://github.com/dotCMS/core/pull/26639/files?diff=unified&w=0#diff-cd183798407f58a8834e971a6eda13bcc1cae9f2fb48244fc111e4a2a8dffa4c))

### CODE REVIEW

**Comment by zJaaal** [comment]
I wonder if we need to make changes in the import, for the JSP, either on the library or directly in the JSP or perhaps both

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-field/dot-edit-content-field.component.spec.ts [hunk 1]
```diff
@@ -22,7 +21,7 @@ import { FIELDS_MOCK, createFormGroupDirectiveMock } from '../../utils/mocks';
 
 // This holds the mapping between the field type and the component that should be used to render it.
 // We need to hold this record here, because for some reason the references just fall to undefined.
-const FIELD_TYPES_COMPONENTS: Record<FIELD_TYPES, Type<unknown>> = {
+const FIELD_TYPES_COMPONENTS: Partial<Record<FIELD_TYPES, Type<unknown>>> = {
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/components/dot-edit-content-field/dot-edit-content-field.component.spec.ts:24]: Why `Partial`?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-binary-field/dot-edit-binary-field.component.ts [hunk 2]
```diff
@@ -42,11 +42,10 @@ import { DotBinaryFieldUrlModeComponent } from './components/dot-binary-field-ur
 import { BinaryFieldMode, BinaryFieldStatus } from './interfaces';
 import { DotBinaryFieldEditImageService } from './service/dot-binary-field-edit-image/dot-binary-field-edit-image.service';
 import { DotBinaryFieldStore } from './store/binary-field.store';
-
-import { getUiMessage } from '../../utils/binary-field-utils';
+import { getUiMessage } from './utils/binary-field-utils';
 
 @Component({
-    selector: 'dot-binary-field',
+    selector: 'dot-edit-binary-field',
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-binary-field/dot-edit-binary-field.component.ts:48]: We are using `dot-edit-content-xxx` as standard, should we do that for this component too?

File: core-web/libs/edit-content/src/lib/components/dot-edit-content-field/dot-edit-content-field.component.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ import { FIELD_TYPES } from '../../models/dot-edit-content-field.enum';
         ReactiveFormsModule,
         DotEditContentFieldsModule,
         DotFieldRequiredDirective,
-        DotBinaryFieldComponent
+        DotEditBinaryFieldComponent
```
Comment (zJaaal) [inline core-web/libs/edit-content/src/lib/components/dot-edit-content-field/dot-edit-content-field.component.ts:32]: You can add this component to the `DotEditContentFieldsModule` that way we centralize the imports/exports of all the fields we create

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-binary-field/dot-edit-binary-field.component.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-binary-field/dot-edit-binary-field.component.ts]: @fmontes Do we want to rename the child components as well? Meaning `dot-binary-field-editor` to `dot-edit-content-binary-field-editor`, and so on?

**Comment by rjvelazco** [comment]
> I wonder if we need to make changes in the import, for the JSP, either in the library or directly in the JSP or perhaps both

The `contenttype-fields-builder` app is the one that builds the `Web-component` we need to make sure to import the `dot-edit-content-binary-field` in that library. That's why we exported the component here:

https://github.com/dotCMS/core/blob/8a3fe84c6a211905567a1655430f24ffb4febc7d/core-web/libs/edit-content/src/index.ts#L2

and imported here: 

https://github.com/dotCMS/core/blob/8a3fe84c6a211905567a1655430f24ffb4febc7d/core-web/apps/contenttype-fields-builder/src/app/app.module.ts#L10-L22

@zJaaal

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Pull request merged**